cookbook/security/securing_application.rst

```diff
@@ -0,0 +1,124 @@
+.. index::
+   single: Security; Best Practices
+
+How to Secure a Symfony Application
+===================================
+
+This article summarizes the steps you should follow to secure your Symfony
+applications.
+
+Configuration
+-------------
+
+Configure the ``secret`` Option
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The value of this option is used to increase the entropy in several security
+related operations. Make sure to change its default value (usually defined in
+the ``app/config/parameters.yml`` file). It's also recommended to change its
+value from time to time.
+
+Read the :ref:`framework.secret documentation <configuration-framework-secret>`
+
+Configure a Strong Password Encryption
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+`Bcrypt`_ is the most recommended password hash algorithm for modern web
+applications. That's why Symfony only recommends to :ref:`use the bcrypt encoder <reference-security-bcrypt>`
+even for in-memory users defined in the ``app/config/security.yml`` file.
+
+Make your Application HTTPS-only
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If your application defines some protected areas under the ``access_control``
+option in ``app/config/security.yml``, use the ``requires_channel`` option as
+explained in :doc:`this article </cookbook/security/force_https>`.
+
+If you prefer to protect some specific routes, define their ``schemes`` routing
+option as explained in :doc:`this article </cookbook/routing/scheme>`.
+
+Enable or Disable the CSRF Protection
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, Symfony forms include a protection against `CSRF attacks`_. In case
+you need to :ref:`disable this protection for some forms <forms-csrf>`, use the
+``csrf_protection`` option. You can even :ref:`disable CSRF protection <reference-csrf_protection-enabled>`
+application-wide.
+
+Furthermore, the form used to log in users doesn't include any CSRF protection.
+Read :doc:`this article </cookbook/security/csrf_in_login_form>` to learn how to
+enable CSRF protection for login forms.
+
+Deployment
+----------
+
+Remove Non-Production Controllers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Symfony's development front controller (``web/app_dev.php``) includes a built-in
+protection to avoid leaking sensitive information in production servers. However,
+when developing custom front controllers is common to forget to add this protection.
+That's why your deployment script or tool should delete any front controller
+different from ``web/app.php``.
+
+Monitoring
+----------
+
+Log Security Events in a Separate File
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For some applications it may be useful to log any security-related event into a
+separate log file. Read :ref:`this article <logging-channel-handler>` to learn
+how to create that ``security.log`` file.
+
+Check Dependencies Vulnerabilities
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. versionadded:: 2.5
```

Comment:
and this section should not be there at all in the doc for the 2.3 branch. It should be added later in a separate PR.